### PR TITLE
Add download of zipped media functionality

### PR DESCRIPTION
**Pull request by nathanbegbie**
See https://praekeltorg.atlassian.net/browse/TUNE-244

### CODE REVIEW

**Comment by nathanbegbie** [comment]
screen shot 2017-01-24 at 2 09 14 pm

screen shot 2017-01-24 at 2 09 58 pm

File: molo/core/utils.py [hunk 1]
```diff
@@ -52,3 +56,21 @@ def generate_slug(text, tail_number=0):
...
         return slug
+
+
+def replace_media_file(file):
+    '''Replace the Current Media Folder.'''
+    media_parent_directory = os.path.dirname(settings.MEDIA_ROOT)
+    zip_file_reference = os.path.join(media_parent_directory, 'new_media.zip')
+
+    if os.path.isdir(settings.MEDIA_ROOT):
+        shutil.rmtree(settings.MEDIA_ROOT)
```
Comment (miltontony) [inline molo/core/utils.py]: This is what I'm concerned about.. did you consider using [copytree](https://docs.python.org/2/distutils/apiref.html#distutils.dir_util.copy_tree)

**Comment by nathanbegbie** [comment]
Better error handling and communication:
screen shot 2017-01-31 at 4 45 23 pm

screen shot 2017-01-31 at 4 45 23 pm
screen shot 2017-01-31 at 4 58 46 pm

File: molo/core/utils.py [hunk 2]
```diff
@@ -58,19 +59,42 @@ def generate_slug(text, tail_number=0):
...
 
-    with zipfile.ZipFile(zip_file_reference, 'r') as z:
-        z.extractall(media_parent_directory)
-
-    os.remove(zip_file_reference)
+    Returns list of files copied accross or
+    raises an exception.
+    '''
+    media_parent_directory = os.path.dirname(settings.MEDIA_ROOT)
+    temp_dir_path = os.path.join(media_parent_directory, 'temp_copy_dir')
```
Comment (miltontony) [inline molo/core/utils.py]: Consider using python's [Temp File](https://docs.python.org/2/library/tempfile.html)
It should make managing the temp file easier

File: molo/core/utils.py [hunk 3]
```diff
@@ -52,3 +58,36 @@ def generate_slug(text, tail_number=0):
     else:
         # No collisions
         return slug
+
+
+def update_media_file(file):
```
Comment (miltontony) [inline molo/core/utils.py]: `file` is a python keyword. Consider using a different name.

**Comment by miltontony** [comment]
👍

**Comment by miltontony** [comment]
🍰 💥

**Pull request merged**